### ISSUE TITLE
Documents SOREL CAN BUS Interface (SCBI)

### ISSUE THREAD

**Issue by hokyswiss** [ddimension/can-ltdc-logger#1]
Hello,

I'm very interested on this project. May you share the SOREL CAN BUS Interface (SCBI) ? Without it, it is very difficult to add functions / develop the interface !

Thanks

**Comment by ghost** [comment]
@hokyswiss I had the same thought and went forward to ask sorel support for the docs. Luckily they had some Customer aimed documentation for me but asked to not publish it any further. Maybe you too ask directly at sorel? If you have further questions contact me anytime. By the way I forked this repo and am in the process to create an scbi-API. My main goal is to support MTDCv5 devices but I'm open for a more general approach.

**Comment by LedariaTim1** [comment]
@aprospero can you tell me if the can interface of the sorel mtdc sends data directly when i connect my raspberry or if i have to send a telegram to the sorel. And can you tell me the bitrate of the CAN bus?

**Comment by ghost** [comment]
@LedariaTim1 at least my mtdc (no SW-Version at hand right now) sends from startup on lots of measurements. If decoding works you should see a stream of CAN datagrams with regular updates at least every 20sec. The bitrate is 250kBaud.
Another important thing in this setup is that you have to take care of CAN-Bus termination. You'll need a resistor of 120Ohms or another device on the second can port of the mtdc. Otherwise the Raspberry may probably have a hard time to decode anythin on the bus.

**Comment by LedariaTim1** [comment]
@aprospero Ok perfect, thanks that helps me a lot, canLow is probably the white wire and CanHigh the brown one, right? (I mean in your fork at the doc on the picture)

**Comment by ghost** [comment]
@LedariaTim1 this photo isn't mine, it's from @ddimension. Honestly, I didn't care which pin is high/low. You can't break anything by accidential swapping them so there is no risk trying and you have a whopping 50% chance to get it right at 1st try! ;)

**Comment by LedariaTim1** [comment]
@aprospero Perfect, communication is now up and running :p I'm getting data with lengths 1, 5, 8 where the 1 length will probably just be a keep alive signal. now I just need to figure out how to understand the received data xD I'll probably contact Sorel support for that as well. Thanks for your help already :D

**Comment by ghost** [comment]
@LedariaTim1 cool! Always happy to help. As stated Sorel has given me some (a bit outdated) info under the premise I won't give it to 3rd party, so go ahead with sorel support! If  you are familiar with c-structs you could also probably figure it out with the info provided in [this](https://github.com/aprospero/can-mtdc-logger/blob/master/code/ctrl/scbi.h) header file of my fork. More info about how to read a message is hidden in [scbi.c](https://github.com/aprospero/can-mtdc-logger/blob/master/code/ctrl/scbi.c).

**Comment by LedariaTim1** [comment]
@aprospero Yep I just wrote an email to support, I'm not 100% up to work with C, but I read your C code a bit, helps a bit :D I want to handle my data in Java and then send it to homeassistant :p

**Comment by ghost** [comment]
@LedariaTim1 godspeed on your journey! If you happen to find quirks/wrongs in the sorel docs (there definitely are) and you are able to sort some out please let me know! Also I have Sorels SBCI Doc Revision 12 from Feb. 16th 2021 - if you get a newer one let me know, too!

**Comment by javier6d2** [comment]
May be some one want to use Tasmota firmware with an ESP32 device to connect to Sorel LTDC Controller: 

[Link Tasmota ](https://tasmota.github.io/docs/Smart-Meter-Interface/#sorel-ltdc-canbus)

**Comment by ghost** [comment]
@javier6d2 I'd be pleased if sorella lib would be of use for anyone. If you want to go ahead please do so - and if I can be of any help, don't hesitate to contact me or do a pull request!
